### PR TITLE
Server/Deployer refactor

### LINKED ISSUE #14: Rename dataflow properties
Deployer is still internally using properties prefixed with `dataflow`. Those should be renamed to `deployer`.

### LINKED ISSUE #15: Make base hdfs dir configurable
Currently we hardcode basedir as `/dataflow`. As a user I'd like that to be configurable.

### LINKED ISSUE #16: Refactor appdeployer package
`dataflow.yarn.appdeployer` should be `deployer.spi.yarn.appdeployer`.

### LINKED ISSUE #17: Javaopts option for appmaster
As a user I'd like to set jvm options for appmaster processes.

### LINKED ISSUE #18: Add deployer auto-config
As a user I'd like to split auto-config from a server so that deployer is more independent.

### PR DESCRIPTION
**Pull request by jvalkeal**
Should go together with https://github.com/spring-cloud/spring-cloud-dataflow-server-yarn/pull/117
- Use shdp snapshots
- Better logging config
- Refactor deployer to be independed from a server
  as much as possible.
- New deployer auto-config
- Add jvm opts for stream appmaster
- Preliminary support for launching new tasks from
  existing streams or tasks.
- Fixes #14
- Fixes #15
- Fixes #16
- Fixes #17
- Fixes #18

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -16,14 +16,16 @@
 
 	<properties>
 		<spring-cloud-deployer.version>1.0.0.RC1</spring-cloud-deployer.version>
-		<spring-hadoop.version>2.4.0.M1</spring-hadoop.version>
+		<spring-cloud-stream.version>1.0.0.RC3</spring-cloud-stream.version>
```
Comment (trisberg) [inline pom.xml]: why RC3? 1.0.2.RELEASE is available.

Comment (trisberg) [inline pom.xml]: also, where is this used?

Comment (jvalkeal) [inline pom.xml]: Ah, that is probably leftover when I had a local merge conflict.  I'll take it away.

**Comment by trisberg** [comment]
Merged via commit 5440be5971d50db0d6e80d2b87015ae78885dd27

**Pull request closed without merging**